src/moscot/problems/base/_base_problem.py

```diff
@@ -474,6 +475,120 @@ def _create_marginals(
     def _estimate_marginals(self, adata: AnnData, *, source: bool, **kwargs: Any) -> ArrayLike:
         return np.ones((adata.n_obs,), dtype=float) / adata.n_obs
 
+    @d.dedent
+    def set_xy(
+        self, data: Union[ArrayLike, pd.DataFrame], tag: Literal["cost", "kernel"], validate_data: bool = True
+    ) -> None:
+        """
+        Set a custom cost matrix/kernel in the linear term.
+
+        Parameters
+        ----------
+        %(data_set)s
+        %(tag)s
+        %(validate_data_set)s
+
+        Returns
+        -------
+        None
+        """
+        if self.problem_kind == ProblemKind.QUAD:
+            logger.info(f"Changing the problem type from {self.problem_kind} to fused-quadratic.")
```

Comment:
I don't think we should allow to change the method from linear to fused quadratic after initialization. Or maybe I'm not seeing it, could you share a use case?

Reply:
A setting would be to have the `TemporalProblem` and then you want to have a prior. Example: Use gene expression in the linear term, and now you want to have ATAC in the quadratic term. The `TemporalProblem` is not constructed to explicitly allow for this setting (as ATAC lives in the same space it is actually not necessary), but the use might want to play around with it.